File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        mx = max(piles)
        l = 1
        r = mx
        result = r

        while(l<=r):
            k = (l+r)//2
            sm = 0

            for i in piles:
                sm += int((i+k-1)/k)
            
            if sm <= h:
                ans = k
                r = k-1
            else:
                l = k+1
        
        return ans

        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mp = 0;
        int l = 0, r = 1;

        while(r<prices.size()){
            if(prices[r] > prices[l]){
                mp = max(mp, prices[r]-prices[l]);
            }else{
                l = r;
            }

            r++;
        }

        return mp;

    }
};
